src/pages/VeSDL/OnChainVote.tsx

```diff
@@ -0,0 +1,254 @@
+
+  const gaugeNames = useMemo(
+    () =>
+      Object.keys(gauges)
+        .filter((lpAddress) => !gauges[lpAddress]?.isKilled)
+        .map((lpAddress) => {
+          const gaugeName = gauges[lpAddress]?.gaugeName
+          const gaugeAddress = gauges[lpAddress]?.address
+          if (gaugeName && gaugeAddress)
+            return {
+              address: gaugeAddress,
+              gaugeName: gaugeName,
+            } as GaugeName
+        }),
+    [gauges],
+  )
+
+  const getVoteUsed = useCallback(async () => {
+    if (account) {
+      const voteUsed = await gaugeControllerContract?.[
+        "vote_user_power(address)"
+      ](account)
+      setVoteUsed(voteUsed?.toNumber())
+    }
+  }, [account, gaugeControllerContract])
+
+  useEffect(() => {
+    if (account && gaugeControllerContract) {
+      gaugeControllerContract.on(
+        "VoteForGauge",
+        (voteDate, fromAddress, gaugeAddress, voteWeight) => {
+          const gaugeName = gaugeNames.find(
+            (gaugeName) =>
+              gaugeName?.address.toLowerCase() ===
+              (gaugeAddress as string).toLowerCase(),
+          )?.gaugeName
+
+          if (
+            gaugeName &&
+            (fromAddress as string).toLowerCase() === account.toLowerCase()
+          ) {
+            setVotes((currentVoteLists) => ({
+              ...currentVoteLists,
+              [gaugeAddress]: {
+                gaugeName,
+                voteDate: (voteDate as BigNumber).toNumber(),
+                weight: voteWeight as BigNumber,
+              },
+            }))
+            void getVoteUsed()
+          }
+        },
+      )
+    }
+    return () => {
+      gaugeControllerContract?.removeAllListeners()
+    }
+  }, [gaugeControllerContract, gaugeNames, votes, getVoteUsed, account])
+
+  const handleVote = async () => {
+    const voteWeightPercent = parseFloat(voteWeightToSubmit)
+    if (!Number.isNaN(voteWeightPercent)) {
+      try {
+        if (veSdlBalance.isZero()) {
+          setAlertMessage("You need veSDL to vote")
+        } else if (selectedGauge?.address && gaugeControllerContract) {
+          const txn = await gaugeControllerContract.vote_for_gauge_weights(
+            selectedGauge.address,
+            Math.round(voteWeightPercent * 100), //convert percent to bps
+          )
+          void enqueuePromiseToast(1, txn.wait(), "vote")
+          setSelectedGauge(null) // Initialize selectedGauge
+          setVoteWeightToSubmit("") // Initialize vote weight input
+        }
+      } catch (error) {
+        console.log("error on vote ==>", error)
+      }
+    }
+  }
+
+  return (
+    <div>
+      <Stack spacing={2} px={2} pt={2} mb={2}>
+        <Typography variant="h2" textAlign="center">
+          {t("voteForNextWeek")}
+        </Typography>
+        <Box display="flex" justifyContent="space-between">
+          <Box display="flex">
+            <Typography component="label">
+              {t("yourVotingPower")}:{" "}
```

Comment:
add padding or margin to Typograph/Box instead of putting manual " " space there.

Reply:
Fixed